src/Api/Controllers/CollectionsController.cs

```diff
@@ -411,8 +485,150 @@ private async Task<bool> CanViewCollectionAsync(Guid orgId, Guid collectionId)
         return false;
     }
 
+    [Obsolete("Pre-Flexible Collections logic. Will be replaced by CollectionsAuthorizationHandler.")]
     private async Task<bool> ViewAtLeastOneCollectionAsync(Guid orgId)
     {
+        DeprecatedPermissionsGuard();
+
         return await _currentContext.ViewAllCollections(orgId) || await _currentContext.ViewAssignedCollections(orgId);
     }
+
+    private async Task<CollectionResponseModel> Get_vNext(Guid collectionId)
+    {
+        var collection = await _collectionRepository.GetByIdAsync(collectionId, _currentContext.UserId.Value);
```

Comment:
By specifying the userId here, it'll get by collectionId _and_ userId, which will only return collections the user is assigned to.

Because we have the permissions check in the auth handler, we don't need this kind of db-level permissions check. So you can just `GetByIdAsync(collectionId)` (which is an overload in the base repository class).

(this is kind of nice! the db call is now super simple.)

Reply:
resolved